integration-tests/relayinterface/chain_components_test.go

```diff
@@ -1024,20 +1041,83 @@ func (it *SolanaChainComponentsInterfaceTester[T]) buildContractReaderConfig(t T
 							Prefix: pdaStructDataPrefix,
 						},
 						OutputModifications: commoncodec.ModifiersConfig{
-							&commoncodec.HardCodeModifierConfig{
-								OnChainValues: map[string]any{
-									"DifferentField":              copy(make([]byte, 32), []byte(testStruct.DifferentField)),
-									"NestedDynamicStruct.Inner.S": copy(make([]byte, 32), []byte(testStruct.NestedDynamicStruct.Inner.S)),
-								},
-								OffChainValues: map[string]any{
-									"ExtraField":                  AnyExtraValue,
-									"DifferentField":              testStruct.DifferentField,
-									"NestedDynamicStruct.Inner.S": testStruct.NestedDynamicStruct.Inner.S,
-								},
+							&commoncodec.AddressBytesToStringModifierConfig{
+								Fields: []string{"AccountStruct.AccountStr"},
+							},
+							&commoncodec.ConstrainedBytesToStringModifierConfig{
+								Fields: []string{"DifferentField", "NestedDynamicStruct.Inner.S"},
+								MaxLen: 32,
+							},
+						},
+					},
+					EventName: {
+						ChainSpecificName: "TestEvent",
+						ReadType:          config.Event,
+						InputModifications: commoncodec.ModifiersConfig{
+							&commoncodec.PropertyExtractorConfig{
+								FieldName:          "Data",
+								EnablePathTraverse: true,
+							},
+							&commoncodec.AddressBytesToStringModifierConfig{
+								Fields:             []string{"AccountStruct.AccountStr"},
+								EnablePathTraverse: true,
+							},
+							&commoncodec.ConstrainedBytesToStringModifierConfig{
+								Fields:             []string{"DifferentField", "NestedDynamicStruct.Inner.S"},
+								EnablePathTraverse: true,
+								MaxLen:             32,
 							},
+						},
```

Comment:
See comments here about why we should fix the bug rather than requiring the user to work around it with complex InputModificaitons:
https://github.com/smartcontractkit/chainlink-solana/pull/1084/files#r1964301442

It's a simple fix, we're just passing true when it should be false... so it gives back the wrong type.  Intentionally requesting the wrong type and then asking the user to modify it to be the right type is bad UI... doing this in the interface test obscures the problem.